### DISCUSSION TITLE
Loading svg in GeoJSONlayer

### DISCUSSION THREAD

**Discussion by TextTL** [visgl/deck.gl#9138] [General]
I tried to use autopacking, and dynamically load svg, and I got this error:

```
parse-to-image-bitmap.ts:57 InvalidStateError: Failed to execute 'createImageBitmap' on 'Window': The image element contains an SVG image without intrinsic dimensions, and no resize options or crop region are specified.
    at safeCreateImageBitmap (parse-to-image-bitmap.ts:55:1)
    at parseToImageBitmap (parse-to-image-bitmap.ts:35:1)
    at async Object.parseImage [as parse] (parse-image.ts:32:1)
    at async parseWithLoader (parse.ts:93:1)
    at async parse (parse.ts:58:1)
    at async load (load.ts:49:1)
...

```
The test SVG is android.svg from https://dev.w3.org/SVG/tools/svgweb/samples/svg-files/android.svg
```

<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 96 105">
  <g fill="#97C024" stroke="#97C024" stroke-linejoin="round" stroke-linecap="round">
    <path d="M14,40v24M81,40v24M38,68v24M57,68v24M28,42v31h39v-31z" stroke-width="12"/>
    <path d="M32,5l5,10M64,5l-6,10 " stroke-width="2"/>
  </g>
  <path d="M22,35h51v10h-51zM22,33c0-31,51-31,51,0" fill="#97C024"/>
  <g fill="#FFF">
    <circle cx="36" cy="22" r="2"/>
    <circle cx="59" cy="22" r="2"/>
  </g>
</svg>
```

The code is 

```
getIcon: (d) => ({
{
        url: "svg/android.svg",
        height: 105,
        width: 96,
    }
})
```

Does anyone know what problem is? I read some posts related to loading svg, and it seems to work for someone; somehow, I couldn't get it to work.

**Comment by Pessimistress** [comment]
https://github.com/visgl/deck.gl/issues/5158

**Reply by Pessimistress** [reply]
I do think the IconLayer can be smarter here, since width/height are already specified by the icon descriptor. This is probably quite easy to implement.

**Reply by TextTL** [reply]
Thank you. We need to either update the document to require svg element to have width and height attributes. It'd be nice to have the loader to use the supplied width and height as well.

**Comment by TextTL** [comment]
got the answer from #5158
